addons/mail/static/src/core/common/attachment_uploader_hook.js

```diff
@@ -157,3 +158,27 @@ export function useAttachmentUploader(thread, { composer, onFileUploaded } = {})
 
     return state;
 }
+
+export const setupAttachment = {
+    prepareFormData(formData, state, file, composer, tmpId, options) {
+        formData.append("thread_id", state.thread.id);
+        formData.append("tmp_url", URL.createObjectURL(file));
+        formData.append("thread_model", state.thread.model);
+        formData.append("is_pending", Boolean(composer));
+        formData.append("temporary_id", tmpId);
+        return formData;
+    },
+    prepareAttachmentData(upload, tmpId, originThread, tmpUrl) {
+        const attachmentData = {
+            filename: upload.title,
+            id: tmpId,
+            mimetype: upload.type,
+            name: upload.title,
+            originThread: originThread,
+            extension: upload.title.split(".").pop(),
+            uploading: true,
+            tmpUrl,
+        };
+        return attachmentData;
+    },
+};
```

Comment:
ESLint is unhappy because `setupAttachment` is used before it's defined.
Also this object is made just for the case of patching. I get why this is introduced, but it's annoying to make code less readable as a result. `setupAttachment.prepareAttachmentData()` is quite obscure without reading the code on what it actually does. "setup" and "prepare" are far too generic wording, in general "setup" is fine as init lifecycle of Component/Service but in other cases this is a poor naming for objects/functions.

I'd suggest to move turn the internal logic of the hook into a class, so that we can patch like we do for classes, and the syntax is nice.


```js
export class AttachmentUploader() {
}
export function useAttachmentUploader() {
     return reactive(new AttachmentUploader());
}
```

That way it's just a `this.method()` that you can patch, you remove `setupAttachment` object which is poorly named (and the class is named similarly to the hook, so naming is good)